docs/content/integrations/snowflake/reference.mdx

```diff
@@ -7,13 +7,61 @@ description: Store your Dagster assets in Snowflake
 
 This reference page provides information for working with [`dagster-snowflake`](/\_apidocs/libraries/dagster-snowflake) features that are not covered as part of the [Using Dagster with Snowflake tutorial](/integrations/snowflake/using-snowflake-with-dagster).
 
+- [Authenticating using a private key](#authenticating-using-a-private-key)
 - [Selecting specific columns in a downstream asset](#selecting-specific-columns-in-a-downstream-asset)
 - [Storing tables in multiple schemas](#storing-tables-in-multiple-schemas)
 - [Using the Snowflake I/O manager with other I/O managers](#using-the-snowflake-io-manager-with-other-io-managers)
 - [Executing custom SQL commands with the Snowflake resource](#executing-custom-sql-commands-with-the-snowflake-resource)
 
 ---
 
+## Authenticating using a private key
+
+In addition to password-based authentication, you can authenticate with Snowflake using a key pair. To set up private key authentication for your Snowflake account, see the instructions in the [Snowflake docs](https://docs.snowflake.com/en/user-guide/key-pair-auth.html#configuring-key-pair-authentication).
+
+You can provide the private key directly to the Snowflake I/O manager using the `private_key` configuration value. If your key is encrypted, you can provide the password using the `private_key_password`.
+
+```python file=/integrations/snowflake/private_key_auth.py startafter=start_direct_key endbefore=end_direct_key
```

Comment:
Thoughts on putting these examples into a tab component? It would visually condense things and allow the user to focus their attention on one thing at a time. Ex:

```
Private keys can either be provided directly to the Snowflake I/O manager or via a file containing the key.

<TabGroup>
<TabItem name="Directly to the I/O manager">
[content]
</TabItem>
<TabItem name="Via a file">
[content]
</TabGroup>
```

I'm not sure if you've used this component before, but I've got a [draft update about it here](https://github.com/dagster-io/dagster/blob/eab3e939e912d719428887a5e7d8e234256b1cf6/docs/README.md#tabs).